### ISSUE TITLE
Update install scripts to support zypper

### ISSUE THREAD

**Issue by r0mant** [gravitational/teleport#29267]
Some customers are using Teleport on SUSE with zypper package manager. zypper can install rpms directly from our existing rpm repository.

Let's update our [automatic upgrades](https://github.com/gravitational/teleport.e/blob/master/assets/systemd-unit-upgrader/rootfs/usr/sbin/teleport-upgrade) and auto-discovery scripts (both [regular](https://github.com/gravitational/teleport/blob/master/api/types/installers/installer.sh.tmpl) and [agentless](https://github.com/gravitational/teleport/blob/master/api/types/installers/agentless-installer.sh.tmpl)) to support zypper when running on SUSE Linux.

**Comment by marcoandredinis** [comment]
Should we update the node join script used during discover wizard as well?

**Comment by r0mant** [comment]
@marcoandredinis Yes good call, updated the description to mention them as well. cc @lxea

**Comment by fspmarshall** [comment]
> zypper can install rpms directly from our existing rpm repository.

note: our package builds for the upgrader define what package manager will be invoked by the upgrade script, so recycling the existing upgrader RPM will result in installing an upgrader that is hardcoded to use `yum`.  We don't want the upgrader "guessing" which package manager to invoke at runtime, as that might lead to unexpected/incorrect behavior.  If having a separate RPM for zypper isn't an option, we'll need to figure out some scheme for detecting the package manager of origin during install time, so that upgraders don't unexpectedly invoke a different package manager than the one that they themselves were installed from 

@fheinecke can probably give better thoughts/advice than I can about the best way to address this.

**Comment by r0mant** [comment]
TBH I don't see a huge problem with the updater script figuring out which package manager to use at runtime (should be fairly trivial to find out whether we're running on SUSE or RHEL) but @fheinecke if you have a better idea on how to handle this, I'm open.

**Comment by r0mant** [comment]
@lxea Can you please finish the backport of https://github.com/gravitational/teleport.e/pull/1876 to branch/v13, test it again on v13 so we can close this.
